src/dist.cpp

```diff
@@ -178,9 +178,7 @@ void zmq::dist_t::distribute (msg_t *msg_)
                 ++i;
             }
         }
-        int rc = msg_->close ();
-        errno_assert (rc == 0);
-        rc = msg_->init ();
+        int rc = msg_->init ();
```

Comment:
It was actually a mistake to close the msg. Msgs must not be closed after writing to pipe, only re-initialized. It is popping up now because before that it was a vsm message (on the stack) so it didn't matter if it was closed. But because it is now might has a reference for a string, it is matters.